libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/MapboxNativeNavigatorImpl.kt

```diff
@@ -480,6 +490,70 @@ object MapboxNativeNavigatorImpl : MapboxNativeNavigator {
     }
 }
 
+private val SUPPORTED_PASSIVE_MANEUVERS = arrayOf(
+    PassiveManeuverType.KTUNNEL_ENTRANCE
+)
+
+private fun RouteInfo?.toRouteInitInfo(): RouteInitInfo? {
+    return if (this != null) {
+        RouteInitInfo(
+            passiveManeuvers
+                .filter { SUPPORTED_PASSIVE_MANEUVERS.contains(it.type) }
+                .map { it.toRouteAlert() }
+        )
+    } else null
+}
+
+private fun List<UpcomingPassiveManeuver>.toUpcomingRouteAlerts(): List<UpcomingRouteAlert> {
+    return this
+        .filter { SUPPORTED_PASSIVE_MANEUVERS.contains(it.maneuver.type) }
+        .map {
+            UpcomingRouteAlert.Builder(it.maneuver.toRouteAlert(), it.distanceToStart).build()
+        }
+}
+
+private fun PassiveManeuver.toRouteAlert(): RouteAlert<*> {
```

Comment:
IMO we should move this to a delegated class where we can have unit tests, independent from the MapboxNativeNavigator

Reply:
@LukasPaczos proposed direction over here https://github.com/mapbox/mapbox-navigation-android/pull/3567

Reply:
Tanks @kmadsen!